src/redux-middleware/scrollCursorIntoView.ts

```diff
@@ -19,6 +114,30 @@ editingValueStore.subscribe(
   }, 400),
 )
 
+// store the previous isPulling value, to only invoke scrollCursorIntoView when it changes to `false`
+let prevIsPulling: boolean | null = null
+/**
+ * Scroll the cursor into view if the sync was the result of navigation to a thought.
+ */
+syncStatusStore.subscribe(state => {
+  if (!state.isPulling && prevIsPulling && !userInteractedAfterNavigation) {
+    scrollCursorIntoView()
+  }
+
+  prevIsPulling = state.isPulling
+})
+
+/**
+ * Whenever the user scrolls, we set `userInteractedAfterNavigation` to true. This prevents `scrollCursorIntoView` from being called
+ * after loading new thoughts.
+ */
+scrollTopStore.subscribe(() => {
+  // Ignore scroll events that are triggered programmatically.
+  if (forcedScrolling) return
```

Comment:
One way to eliminate `forcedScrolling` would be to unsubscribe from `scrollTopStore` on `startForcedScrolling`, and then resubscribe after the 50ms. That would reduce the amount of state, but would add complexity to the subscription.

Seems to be about the same level of complexity all things considered.